### ISSUE TITLE
SourceCodeGenerator generates logs - where are the log levels set?

### ISSUE THREAD

**Issue by rbellamy** [slick/slick#1666]
I'm working the the 

I've customized the `SourceCodeGenerator`:
```
class VintageGenerator(val profile: JdbcProfile, url: String, jdbcDriver: String, user: String, password: String) {
  val dbFactory = profile.api.Database
  val db = dbFactory.forURL(url, driver = jdbcDriver, user = user, password = password, keepAliveConnection = true)
...
  val scg = db.run(model).map { model =>
    new SourceCodeGenerator(model) {}
  }
}

I have neither a `logback.xml` nor a `logback-test.xml` file. I'm seeing a slew of `BasicAction.action` debug messages:
```
[info] Running com.terradatum.dao.vintage.VintageGenerator com.terradatum.slick.jdbc.ExtendedOracleProfile oracle.jdbc.driver.OracleDriver jdbc:oracle:thin:@someserver someusername somepassword /home/rbellamy/Development/Terradatum/aergo/aergo-vintage-dao/target/scala-2.11/src_managed/main/slick com.terradatum.dao.vintage
21:22:14.988 [run-main-0] INFO  c.t.slick.jdbc.ExtendedOracleProfile -  >>> binding SDO_GEOMETRY -> com.terradatum.math.Vec2 
21:22:15.206 [run-main-0] DEBUG slick.basic.BasicBackend.action - #1: StreamingResultAction []
21:22:15.372 [AsyncExecutor.default-1] DEBUG slick.jdbc.DriverDataSource - Driver oracle.jdbc.driver.OracleDriver not already registered; trying to load it
21:22:15.374 [AsyncExecutor.default-1] DEBUG slick.jdbc.DriverDataSource - Loaded driver oracle.jdbc.driver.OracleDriver but it did not register with DriverManager; trying to instantiate directly
21:22:17.476 [ForkJoinPool-1-worker-29] DEBUG slick.basic.BasicBackend.action - #2: success Vector(MTable(MQName(METRICS.AGENT),TABLE,null,None,None,None), MTable(MQName(METRICS.BROKER),TABLE,null,None,None,None), MTable(MQName(METRICS.GEO),TABLE,null,None,None,None), MTable(MQName(METRICS.MLS),TABLE,null,None,None,None), MTable(MQName(METRICS.OFFICE),TABLE,null,None,None,None), MTable(MQName(METRICS.PROPERTY),TABLE,null,None,None,None), MTable(MQName(METRICS.SIDES),TABLE,null,None,None,None))
...
```

I can see where a new `SlickLogger` is being created in the `BasicBackend`:
```
trait BasicBackend { self =>
  protected lazy val actionLogger = new SlickLogger(LoggerFactory.getLogger(classOf[BasicBackend].getName+".action"))
```

As well, I know that the `Logging` trait is applied to the profile hierarchy.

... and then of course, `def isDebugEnabled = slf4jLogger.isDebugEnabled()`

I've read the [Logging](http://slick.lightbend.com/doc/3.2.0-M1/config.html#logging) documentation, and I've looked for errant system properties or environment variables that could account for this unexpected log output. But I cannot for the life of me find where the the log level is being set.

**Comment by cvogt** [comment]
@rbellamy actually I am not totally sure, but seems like without a logback.xml the default log level *info* is used, which can be overridden by a logback.xml, which is read by logback itself. So by putting a logback.xml onto your classpath you can probably solve this.

@szeiger Any additional thoughts?

**Comment by rbellamy** [comment]
@cvogt yep - however, following the custom codegen example, the codegen is happening in a sub-module of the larger project, and therefore I had to put the `logback.xml` in that module's classpath. That indeed acts as a workaround - I should have mentioned that I was able to adjust the codegen logging using this method.

I really want to understand the inner workings of slick, hence my question of what is going on WITHOUT a logback configuration...

**Comment by rktoomey** [comment]
Been troubleshooting this log output for the better part of a week now....  Finally saw this issue.  We are getting outright `DEBUG` logging, and it's taking up a lot of console real estate 😿 .

Humbly suggest that a default log level of `WARN` or `ERROR` is more suitable to most ordinary human being use cases.
